packages/app/src/components/time-series-chart/components/week-numbers.tsx

```diff
@@ -0,0 +1,187 @@
+import { getWeekInfo } from '~/components/stacked-chart/logic';
+import { useIntl } from '~/intl';
+import { colors } from '~/style/theme';
+import { replaceVariablesInText } from '~/utils/replace-variables-in-text';
+import { useUniqueId } from '~/utils/use-unique-id';
+import { Bounds } from '../logic';
+
+/**
+ * Only show this amount of week numbers
+ */
+const maximumWeekCount = 6;
+
+const DAY_IN_SECONDS = 24 * 60 * 60;
+const WEEK_IN_SECONDS = 7 * DAY_IN_SECONDS;
+
+interface WeekNumberProps {
+  startUnix: number;
+  endUnix: number;
+  xScale: ScaleLinear<number, number> | ScaleBand<number>;
+  bounds: Bounds;
+  formatXAxis: (date_unix: number) => string;
+}
+
+export function WeekNumbers({
+  startUnix,
+  endUnix,
+  xScale,
+  bounds,
+  formatXAxis,
+}: WeekNumberProps) {
+  /**
+   * Used for the clip path,
+   * which prevents grid lines and axis from bleeding out of the graph
+   */
+  const id = useUniqueId();
+  const { siteText } = useIntl();
+
+  const { weekGridLines, weekDateLabels, weekNumbersLabels } = useMemo(
+    () => calculateWeekNumberAxis(startUnix, endUnix),
+    [startUnix, endUnix]
+  );
+  const weekRenderWidth = getWeekRenderWidth(
+    xScale(weekGridLines[2]),
+    xScale(weekGridLines[1])
+  );
+
+  const formatWeekNumberAxis = useCallback(
+    (dateUnix: number) => {
+      const date = new Date(dateUnix * 1000);
+      const weekInfo = getWeekInfo(date);
+      return replaceVariablesInText(siteText.common.week_number_label, {
+        weekNumber: weekInfo.weekNumber,
+      });
+    },
+    [siteText.common]
+  );
+
+  return (
+    <>
+      <RectClipPath
+        id={id}
+        width={bounds.width}
+        height={bounds.height + 200}
+        x={0}
+        y={-100}
+      />
+      <g
+        css={css({
+          clipPath: `url(#${id})`,
+        })}
+      >
+        <GridColumns
+          height={bounds.height}
+          scale={xScale}
+          numTicks={weekGridLines.length}
+          tickValues={weekGridLines}
+          stroke={colors.lightGray}
+          width={bounds.width}
+          strokeDasharray="4 2"
+        />
+
+        <AxisBottom
+          scale={xScale}
+          tickValues={weekDateLabels}
+          tickFormat={formatXAxis as AnyTickFormatter}
+          top={bounds.height}
+          stroke={colors.silver}
+          tickLabelProps={() => ({
+            fill: colors.data.axisLabels,
+            fontSize: 12,
+            textAnchor: 'middle',
+          })}
+          hideTicks
+        />
+
+        <AxisTop
+          scale={xScale}
+          tickValues={weekNumbersLabels}
+          tickFormat={formatWeekNumberAxis as AnyTickFormatter}
+          stroke={colors.silver}
+          hideTicks
+          tickLabelProps={() => ({
+            fill: colors.data.axisLabels,
+            fontSize: 12,
+            textAnchor: 'middle',
+            transform: `translate(${weekRenderWidth / 2} 0)`,
```

Comment:
aren't you missing a css unit here? 
```suggestion
            transform: `translate(${weekRenderWidth / 2}px 0)`,
```

Reply:
I think it defaults to SVG units here